src/main/scala/cromwell/engine/workflow/WorkflowActor.scala

```diff
@@ -369,4 +399,11 @@ case class WorkflowActor(workflow: WorkflowDescriptor,
   }
 
   private def isWorkflowDone: Boolean = executionStore.forall(_._2 == ExecutionStatus.Done)
+
+  def isExecutionStateFinished(es: ExecutionStatus): Boolean = {
+    val terminalStates = List(ExecutionStatus.Failed, ExecutionStatus.Done, ExecutionStatus.Aborted)
```

Comment:
Also prefer `Vector` over `List` unless there's a good reason not to (and they def do exist)